src/wazuh_modules/wm_database.c

```diff
@@ -471,6 +499,88 @@ void wm_sync_multi_groups(const char *dirname) {
     wdb_update_groups(dirname, &wdb_wmdb_sock);
 }
 
+void wm_sync_legacy_groups_files() {
+    DIR *dir = opendir(GROUPS_DIR);
+
+    if (!dir) {
+        mdebug1("Couldn't open directory '%s': %s.", GROUPS_DIR, strerror(errno));
+        return;
+    }
+
+    mtdebug1(WM_DATABASE_LOGTAG, "Scanning directory '%s'.", GROUPS_DIR);
+
+    struct dirent *dir_entry = NULL;
+    int sync_result = OS_INVALID;
+    char group_file_path[OS_SIZE_512] = {0};
+    bool is_dir_empty = true;
+
+    while ((dir_entry = readdir(dir)) != NULL) {
+        if (dir_entry->d_name[0] != '.') {
+            is_dir_empty = false;
+            snprintf(group_file_path, OS_SIZE_512, "%s/%s", GROUPS_DIR, dir_entry->d_name);
+
+            if (is_worker) {
+                mdebug1("Group file '%s' won't be synced in a worker node, removing.", group_file_path);
+                unlink(group_file_path);
+            } else {
+                sync_result = wm_sync_group_file(dir_entry->d_name, group_file_path);
+
+                if (OS_SUCCESS == sync_result) {
+                    mdebug1("Group file '%s' successfully synced, removing.", group_file_path);
+                    unlink(group_file_path);
+                } else {
+                    merror("Failed during the groups file '%s' syncronization.", group_file_path);
+                }
+            }
+        }
+    }
+    closedir(dir);
+
+    if (is_dir_empty) {
```

Comment:
I understand that this logic will work by the second time the manager is started. 
As this isn´t a common scenario, the users can get confused to find the legacy groups folder as empty after the upgrade, we should find a way to delete it (if empty) after the first restart.

Reply:
We can do again the
```
while ((dir_entry = readdir(dir)) != NULL) {
        if (dir_entry->d_name[0] != '.') {
```
logic after the files synchronization.
Or we can just move the 
`is_dir_empty = false;`
to the unique place where the file isn´t deleted, in the error of the synchronization.
In this way, the folder will be deleted in the first attempt.

Reply:
The second approach is more elegant. Now, the folder is deleted in the first try if the synchronization doesn't fail.

Thank you !